packages/application-generic/src/encryption/encrypt-provider.ts

```diff
@@ -41,19 +44,34 @@ export function decryptCredentials(
 
   for (const key in credentials) {
     decryptedCredentials[key] =
-      typeof credentials[key] === 'string' &&
-      isEncryptedCredential(credentials[key])
-        ? decryptProviderSecret(credentials[key])
+      typeof credentials[key] === 'string' && novuEncrypted(credentials[key])
+        ? decryptSecret(credentials[key])
         : credentials[key];
   }
 
   return decryptedCredentials;
 }
 
-function isEncryptedCredential(text: string): boolean {
+export function encryptApiKey(apiKey: string): EncryptedSecret {
+  if (novuEncrypted(apiKey)) {
+    return apiKey;
+  }
+
+  return encryptSecret(apiKey);
+}
+
+export function decryptApiKey(apiKey: string): string {
+  if (novuEncrypted(apiKey)) {
+    return decryptSecret(apiKey);
+  }
+
+  return apiKey;
+}
```

Comment:
those are middleware for the encryption-decryption that will help us make sure that we won't encrypt/decrypt twice for no reason.

Reply:
shouldn't we have these checks actually in the `encryptSecret, decryptSecret` functions? because I see that they are exposed and used in other places.

Reply:
That is a good point but i did not want to merge the base functions `encryptSecret, decryptSecret` that at the moment responsible for adding and removing the prefix.  as it could affect the integration credentials as well.